sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/AggUtils.scala

```diff
@@ -135,19 +135,25 @@ object AggUtils {
     }
     val distinctAttributes = namedDistinctExpressions.map(_.toAttribute)
     val groupingAttributes = groupingExpressions.map(_.toAttribute)
+    val filterWithDistinctAttributes = functionsWithDistinct.flatMap(_.filterAttributes.toSeq)
 
     // 1. Create an Aggregate Operator for partial aggregations.
     val partialAggregate: SparkPlan = {
       val aggregateExpressions = functionsWithoutDistinct.map(_.copy(mode = Partial))
       val aggregateAttributes = aggregateExpressions.map(_.resultAttribute)
       // We will group by the original grouping expression, plus an additional expression for the
-      // DISTINCT column. For example, for AVG(DISTINCT value) GROUP BY key, the grouping
-      // expressions will be [key, value].
+      // DISTINCT column and the referred attributes in the FILTER clause associated with each
+      // aggregate function. For example:
+      // 1.for the AVG (DISTINCT value) GROUP BY key, the grouping expression will be [key, value];
+      // 2.for AVG (DISTINCT value) Filter (WHERE value2> 20) GROUP BY key, the grouping expression
```

Comment:
This might be more efficient, but I don't think it worth to touch `planAggregateWithOneDistinct` which is really convoluted.

Can we handle single distinct with filter in `RewriteDistinctAggregates`?